### PR TITLE
Coverting Smart Banner to Styled Components [WIP]

### PR DESCRIPTION
**Pull request by bahaa96**
### Checklist

- [ ] Unit tests written
- [x] Manually tested
- [ ] Prettier & ESLint were run
- [ ] New dependencies are included in `package-lock.json`

### CODE REVIEW

File: src/components/SmartBanner/index.tsx [hunk 1]
```diff
@@ -238,39 +244,34 @@ class SmartBanner extends Component<Props> {
...
-    const wrapperClassName = `smartbanner smartbanner-${deviceType}`;
-    const iconStyle = {
-      backgroundImage: `url(${icon})`,
-    };
 
     return (
-      <div className={wrapperClassName}>
-        <div className="smartbanner-container">
-          <button
+      <Container className={`smartbanner ${deviceType}`}>
```
Comment (mmahalwy) [inline src/components/SmartBanner/index.tsx]: is class `smartbanner` needed?

File: src/styles/main.global.scss [hunk 2]
```diff
@@ -142,6 +141,14 @@ li em {
   overflow-y: auto;
 }
 
+.smartbanner-show {
```
Comment (mmahalwy) [inline src/styles/main.global.scss]: these should not be needed no?

Comment (bahaa96) [reply src/styles/main.global.scss]: I guess this is the class the shows up the smartbanner

Comment (bahaa96) [reply src/styles/main.global.scss]: this class being given to the element from Show() and Hide() methods in the smartbanner container

File: src/components/SmartBanner/index.tsx (hunk 1 above)
Comment (bahaa96) [reply src/components/SmartBanner/index.tsx]: it's needed to get controlled by the CSS class .smartbanner-show

File: src/styles/main.global.scss (hunk 2 above)
Comment (mmahalwy) [reply src/styles/main.global.scss]: Maybe have that be a prop to `Container`?

Comment (mmahalwy) [reply src/styles/main.global.scss]: Hmm.. I guess I am lost where this is used :S

Comment (bahaa96) [reply src/styles/main.global.scss]:
```
hide = () => {
    if (window.document && window.document.querySelector('html')) {
      (window.document.querySelector('html') as Element).classList.remove(
        'smartbanner-show'
      );
    }
  };

  show = () => {
    if (window.document && window.document.querySelector('html')) {
      (window.document.querySelector('html') as Element).classList.add(
        'smartbanner-show'
      );
    }
  };
```

Comment (mmahalwy) [reply src/styles/main.global.scss]: Right. The essence of this needs to be corrected. If we wish to have some margin to leave space for the smart banner, then we bought to add this to `App.tsx`. Changing the html tag in React seems like a solution that will create bugs or unsustainable.

Comment (bahaa96) [reply src/styles/main.global.scss]: So you need me to make based on a state ?

**Pull request closed without merging**